### ISSUE TITLE
[Feature] Save cache from requests and load

### ISSUE THREAD

**Issue by SinanAkkoyun** [sgl-project/sglang#1932]
### Checklist

- [ ] 1. If the issue you raised is not a feature but a question, please raise a discussion at https://github.com/sgl-project/sglang/discussions/new/choose Otherwise, it will be closed.
- [X] 2. Please use English, otherwise it will be closed.

### Motivation

This might be difficult to implement, but I am facing the following issue:
When running Qwen2-VL on bigger images, the preprocessor takes a long time to convert the images to tokens.

It would be awesome if we could have a way (OpenAI API with extra parameters) to tell the backend to store the cache of a request and load it by ID for another request, which would make it possible to not reprocess every image (and prompt in general) on each call.

If my problem could be solved in an easier way I would be thankful for any input :)

**Comment by merrymercy** [comment]
Yes, this seems to be a good feature to have. Contributions are welcome!

**Comment by Tushar-ml** [comment]
Do you know if semantic cache will work here? 
Resources:
- https://www.redisvl.com/user_guide/llmcache_03.html
- https://redis.io/docs/latest/integrate/redisvl/user-guide/semantic-caching/
